pennylane/beta/tapes/tape.py

```diff
@@ -0,0 +1,835 @@
+        >>> tape.get_parameters()
+        [0.543]
+
+        The ``trainable_only`` argument can be set to ``False`` to instead return
+        all parameters:
+
+        >>> tape.get_parameters(trainable_only=False)
+        [0.432, 0.543, 0.133]
+        """
+        params = []
+        iterator = self.trainable_params if trainable_only else self._par_info
+
+        for p_idx in iterator:
+            op = self._par_info[p_idx]["op"]
+            op_idx = self._par_info[p_idx]["p_idx"]
+            params.append(op.data[op_idx])
+
+        return params
+
+    def set_parameters(self, params, trainable_only=True):
+        """Set the parameters incident on the tape operations.
+
+        Args:
+            params (list[float]): A list of real numbers representing the
+                parameters of the quantum operations. The parameters should be
+                provided in order of appearance in the quantum tape.
+            trainable_only (bool): if True, set only trainable parameters
+
+        **Example**
+
+        .. code-block:: python
+
+            from pennylane.beta.tapes import QuantumTape
+            from pennylane.beta.queuing import expval, var, sample, probs
+
+            with QuantumTape() as tape:
+                qml.RX(0.432, wires=0)
+                qml.RY(0.543, wires=0)
+                qml.CNOT(wires=[0, 'a'])
+                qml.RX(0.133, wires='a')
+                expval(qml.PauliZ(wires=[0]))
+
+        By default, all parameters are trainable and can be modified:
+
+        >>> tape.set_parameters([0.1, 0.2, 0.3])
+        >>> tape.get_parameters()
+        [0.1, 0.2, 0.3]
+
+        Setting the trainable parameter indices will result in only the specified
+        parameters being modifiable. Note that this only modifies the number of
+        parameters that must be passed.
+
+        >>> tape.trainable_params = {0, 2} # set the first and third parameter as free
+        >>> tape.set_parameters([-0.1, 0.5])
+        >>> tape.get_parameters(trainable_only=False)
+        [-0.1, 0.2, 0.5]
+
+        The ``trainable_only`` argument can be set to ``False`` to instead set
+        all parameters:
+
+        >>> tape.set_parameters([4, 1, 6], trainable_only=False)
+        >>> tape.get_parameters(trainable_only=False)
+        [4, 1, 6]
+        """
+        if trainable_only:
+            iterator = zip(self.trainable_params, params)
+            required_length = self.num_params
+        else:
+            iterator = enumerate(params)
+            required_length = len(self._par_info)
+
+        if len(params) != required_length:
+            raise ValueError("Number of provided parameters does not match.")
+
+        for idx, p in iterator:
+            op = self._par_info[idx]["op"]
+            op.data[self._par_info[idx]["p_idx"]] = p
+
+    # ========================================================
+    # properties, setters, and getters
+    # ========================================================
```

Comment:
isn't the ``trainable_params`` function above also a setter?

Reply:
Good point, I'll change this